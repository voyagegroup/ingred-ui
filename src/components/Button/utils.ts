import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
