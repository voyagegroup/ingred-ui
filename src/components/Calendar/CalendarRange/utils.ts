import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
