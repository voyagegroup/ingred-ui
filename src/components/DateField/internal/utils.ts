export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
