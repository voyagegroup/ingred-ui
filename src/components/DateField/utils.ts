import { useMemo } from "react";

type Sections = {
  start: number;
  end: number;
  value: string;
  editable: boolean;
};

/**
 * 何らかのフォーマットで入ってくる日付を開始位置と終了位置と値を持つセクションに分割する
 * useDateField で format された日付操作を汎用的に行うために必要なプロパティを返す
 * 例) 2023-01-02 -> [
 *   { start: 0, end: 3, value: "2023", editable: true },
 *   { start: 4, end: 4, value: "-", editable: false },
 *   { start: 5, end: 6, value: "01", editable: true },
 *   { start: 7, end: 7, value: "-", editable: false },
 *   { start: 8, end: 9, value: "02", editable: true }
 * ]
 *
 * @param formattedDate 何らかのフォーマットで入ってくる日付
 * @returns 開始位置と終了位置と値を持つセクション
 */
export const getSections = (formattedDate?: string | null) => {
  if (!formattedDate) {
    return [];
  }

  const sections: Sections[] = [];
  let start = 0;
  let isPrevCharDigit = !isNaN(Number(formattedDate[0]));

  for (let index = 1; index <= formattedDate.length; index++) {
    const currentChar = formattedDate[index];
    const isCurrentCharDigit =
      !isNaN(Number(currentChar)) && currentChar !== " ";

    if (
      isCurrentCharDigit !== isPrevCharDigit ||
      currentChar === " " ||
      index === formattedDate.length
    ) {
      sections.push({
        start,
        end: index - 1,
        value: formattedDate.slice(start, index),
        editable: isPrevCharDigit,
      });
      start = index;
      isPrevCharDigit = isCurrentCharDigit;
    }
  }

  return sections;
};

/**
 * 開始位置と終了位置と値を持つセクションをフォーマットされた日付に変換する
 */
export const formatString = (sectionsWithCharactor: Sections[]) =>
  sectionsWithCharactor.map((section) => section.value).join("");

type ReactRef<T> =
  | React.RefCallback<T>
  | React.MutableRefObject<T>
  | React.ForwardedRef<T>
  | string
  | null
  | undefined;

// from: https://github.com/voyagegroup/ingred-ui/blob/master/src/hooks/useMergeRefs.ts
export function useMergeRefs<T>(...refs: ReactRef<T>[]): React.Ref<T> {
  return useMemo(() => {
    if (refs.every((ref) => ref === null)) {
      return null;
    }
    return (refValue: T) => {
      for (const ref of refs) {
        if (typeof ref === "function") {
          ref(refValue);
        } else if (ref && typeof ref !== "string") {
          ref.current = refValue;
        }
      }
    };
  }, [refs]);
}
