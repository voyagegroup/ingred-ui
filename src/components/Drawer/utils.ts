/**
 * 相対値をpx値に変換するヘルパー関数
 * @param value - 変換する値（数値またはvw/vh/%/px等の単位付き文字列）
 * @param direction - Drawerの方向（変換時の基準値決定に使用）
 * @returns px値（数値）
 */
export const convertToPixels = (value: string | number, direction: "left" | "right" | "bottom"): number => {
  if (typeof value === "number") {
    return value; // 数値はpx扱い
  }
  
  // 文字列の場合は単位チェック
  const numericValue = parseFloat(value);
  
  if (value.includes("vw")) {
    return (numericValue / 100) * window.innerWidth;
  }
  if (value.includes("vh")) {
    return (numericValue / 100) * window.innerHeight;
  }
  if (value.includes("%")) {
    // %の場合は親要素基準だが、ここではviewport基準として扱う
    const isHorizontal = direction === "left" || direction === "right";
    return (numericValue / 100) * (isHorizontal ? window.innerWidth : window.innerHeight);
  }
  if (value.includes("px")) {
    return numericValue;
  }
  
  // 単位なしの文字列は警告を出す
  if (!isNaN(numericValue)) {
    console.warn(`Drawer: 文字列サイズには単位を指定してください。"${value}" → "${value}px" を推奨`);
    return numericValue; // 一応数値として扱う
  }
  
  // 無効な値の場合はエラー警告とフォールバック
  console.error(`Drawer: 無効なサイズ値です: "${value}". デフォルト値を使用します。`);
  return 400; // フォールバック
}; 