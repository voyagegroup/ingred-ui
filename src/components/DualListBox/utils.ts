import {
  CandidateItem,
  DualListBoxCandidateItem,
  DualListBoxItem,
} from "./DualListBox";

export const getCandidateItems = (
  candidateItems: DualListBoxCandidateItem[],
  selectedItems: DualListBoxItem[],
) => {
  const result: CandidateItem[] = [];
  candidateItems.forEach((candidateItem) => {
    if (candidateItem.items) {
      result.push({
        ...candidateItem,
        items: getCandidateItems(candidateItem.items, selectedItems),
      });
      return;
    }

    const targetItem = selectedItems.find(
      (selectedItem) => selectedItem.id === candidateItem.id,
    );
    if (targetItem) {
      result.push({
        ...candidateItem,
        selected: true,
      });
    } else {
      result.push(candidateItem);
    }
  });

  return result;
};
