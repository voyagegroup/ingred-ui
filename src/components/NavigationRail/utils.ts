import * as React from "react";

export type NavigationRailContextValues = {
  isOpen: boolean;
  isFixed: boolean;
  handleOpen?: () => void;
  handleClose?: () => void;
  handleFixed?: () => void;
  handleUnFixed?: () => void;
};

export const NavigationRailContext =
  React.createContext<NavigationRailContextValues>({
    isOpen: false,
    isFixed: false,
  });
