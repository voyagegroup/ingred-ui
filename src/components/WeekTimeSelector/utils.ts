export const hex2bin = (hex: string) => {
  if (!hex) return Array(7).fill(Array(24).fill("1")).join("");

  let out = "";
  for (const c of hex) {
    switch (c) {
      case "0":
        out += "0000";
        break;
      case "1":
        out += "0001";
        break;
      case "2":
        out += "0010";
        break;
      case "3":
        out += "0011";
        break;
      case "4":
        out += "0100";
        break;
      case "5":
        out += "0101";
        break;
      case "6":
        out += "0110";
        break;
      case "7":
        out += "0111";
        break;
      case "8":
        out += "1000";
        break;
      case "9":
        out += "1001";
        break;
      case "a":
        out += "1010";
        break;
      case "b":
        out += "1011";
        break;
      case "c":
        out += "1100";
        break;
      case "d":
        out += "1101";
        break;
      case "e":
        out += "1110";
        break;
      case "f":
        out += "1111";
        break;
      default:
        return "";
    }
  }

  return out;
};

const bin2hex = (bin: string) => {
  let out = "";
  for (let i = 0; i < bin.length; i += 4) {
    const c = bin.substr(i, 4);
    switch (c) {
      case "0000":
        out += "0";
        break;
      case "0001":
        out += "1";
        break;
      case "0010":
        out += "2";
        break;
      case "0011":
        out += "3";
        break;
      case "0100":
        out += "4";
        break;
      case "0101":
        out += "5";
        break;
      case "0110":
        out += "6";
        break;
      case "0111":
        out += "7";
        break;
      case "1000":
        out += "8";
        break;
      case "1001":
        out += "9";
        break;
      case "1010":
        out += "a";
        break;
      case "1011":
        out += "b";
        break;
      case "1100":
        out += "c";
        break;
      case "1101":
        out += "d";
        break;
      case "1110":
        out += "e";
        break;
      case "1111":
        out += "f";
        break;
      default:
        return "";
    }
  }

  return out;
};

/**
 * @param hex odr_targeting_week_time の16進数表記
 * @returns 週ごとの時間帯ターゲティングの設定の多重配列。[曜日][時間]の順（0番目が月曜日、0番目のn番目がn時）
 */
export const getTargetSetting = (hex?: string): string[][] => {
  if (!hex) return Array(7).fill(Array(24).fill("1"));

  const binary = hex2bin(hex);
  const times: string[][] = [];

  for (let d = 0; d < 7; d++) {
    times[d] = [];
    for (let h = 0; h < 24; h++) {
      times[d][h] = binary.charAt(d * 24 + h);
    }
  }

  return times;
};

export const convertTargetSettingToHex = (targetSetting: string[][]) => {
  let bin = "";
  for (let d = 0; d < 7; d++) {
    for (let h = 0; h < 24; h++) {
      bin += targetSetting[d][h];
    }
  }

  return bin2hex(bin);
};

/**
 * 開始位置と終了位置のそれぞれのの曜日と時間帯から、選択範囲を取得する
 */
export const getNewWeekTimeList = (
  startWeekIndex: number,
  startTimeIndex: number,
  endWeekIndex: number,
  endTimeIndex: number,
  newValue: string,
  weekTimeList: string[][],
) => {
  const newWeekTimeList = [...weekTimeList];
  const minWeekIndex = Math.min(startWeekIndex, endWeekIndex);
  const maxWeekIndex = Math.max(startWeekIndex, endWeekIndex);
  const minTimeIndex = Math.min(startTimeIndex, endTimeIndex);
  const maxTimeIndex = Math.max(startTimeIndex, endTimeIndex);

  for (let i = minWeekIndex; i <= maxWeekIndex; i++) {
    const weekTime = newWeekTimeList[i];
    const newWeekTime = weekTime.map((t, index) => {
      if (
        i >= minWeekIndex &&
        i <= maxWeekIndex &&
        index >= minTimeIndex &&
        index <= maxTimeIndex
      ) {
        return newValue; // 選択範囲内のセルの場合、newValueに更新
      }
      return t;
    });

    newWeekTimeList[i] = newWeekTime;
  }

  return newWeekTimeList;
};
