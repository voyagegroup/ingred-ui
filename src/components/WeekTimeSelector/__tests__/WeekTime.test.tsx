import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import WeekTime from "..";

describe("WeekTime component testing", () => {
  afterEach(cleanup);

  test("WeekTime", () => {
    const { asFragment } = renderWithThemeProvider(
      <WeekTime
        weekTime="ffffffffffffffffffffffffffffffffffffffffff"
        onChange={jest.fn()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
