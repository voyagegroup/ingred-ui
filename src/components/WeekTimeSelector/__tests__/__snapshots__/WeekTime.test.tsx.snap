// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekTime component testing WeekTime 1`] = `
<DocumentFragment>
  <div
    class="sc-eCYdqJ"
  >
    <div
      class="sc-jSMfEi iJRLpY"
    >
      <div
        class="sc-gKXOVf sc-papXJ hVDQsL eYQsLl"
      />
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        0
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        1
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        2
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        3
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        4
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        5
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        6
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        7
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        8
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        9
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        10
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        11
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        12
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        13
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        14
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        15
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        16
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        17
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        18
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        19
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        20
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        21
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        22
      </div>
      <div
        class="sc-gKXOVf sc-ftvSup hVDQsL foQQeW"
      >
        23
      </div>
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Sun
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Mon
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Tue
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Wed
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Thu
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Fri
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <div
        class="sc-gKXOVf sc-iBkjds hVDQsL feTYsA"
      >
        Sat
      </div>
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
      <button
        class="sc-jqUVSM cxmRec"
      />
    </div>
  </div>
</DocumentFragment>
`;
